source/jormungandr/jormungandr/interfaces/v1/ResourceUri.py

```diff
@@ -172,25 +172,23 @@ def __init__(self, resource):
         self.resource = resource
     def complete(self, data, collect):
         queue = deque()
-        result = []
-        for v in data.itervalues():
-            queue.append(v)
+        result = deque()
+        queue.extend(data.itervalues())
         collect_type = collect["type"]
+        del_types = collect["del"]
         while queue:
             elem = queue.pop()
             if isinstance(elem, (list, tuple)):
                 queue.extend(elem)
             elif hasattr(elem, 'iterkeys'):
-                if 'type' in elem and elem['type'] == collect_type:
+                if elem.get('type', None) == collect_type:
                     if collect_type == "notes":
                         result.append({"id": elem['id'], "value": elem['value'], "type": collect_type})
                     elif collect_type == "exceptions":
-                        type_= "Remove"
-                        if elem['except_type'] == 0:
-                            type_ = "Add"
+                        type_ = "Add" if not elem['except_type'] else "Remove"
                         result.append({"id": elem['id'], "date": elem['date'], "type" : type_})
-                    for del_ in collect["del"]:
-                        del elem[del_]
+
+                    [elem.pop(del_) for del_ in del_types]
```

Comment:
I'd have used `map`. After a little perf test with `python2.7`,  list comprehension(`[]`) is very little tiny faster than the 'for loop', but the `map` is 3 times faster...